docs/design/fsdp-design-en.md

```diff
@@ -0,0 +1,373 @@
+# DLRover's FSDP Elastic Training Design Document
+
+## 1. Background
+
+As the scale of machine learning models has grown, the necessity for distributed
+training has become paramount. DLRover is especially proficient in addressing
+challenges such as node failures, dynamic resource allocation, and automatic
+recovery. While PyTorch provides support through FSDP (Fully Sharded Data Parallel),
+DLRover still faces specific issues related to dynamic resources and fault tolerance.
+These problems often manifest as OOM (Out of Memory) errors and complications in
+reloading sharded checkpoints when the world size alters."
+
+### 1.1 Motivation
+
+DLRover aims to support FSDP at the system level for:
+
+1. **System-Level FSDP Elastic Training:** DLRover should recognize FSDP
+training, aiding users in auto-saving and loading checkpoints.
+2. **Enhanced PyTorch FSDP Resharding:** PyTorch's save method requires
+refinement for dynamic resharding when world size varies.
+
+### 1.2 Main Challenges
+
+FSDP implementation in DLRover presents hurdles:
+
+1. **System-Level Interface for FSDP Elastic Training:** The
+`ElasticTrainer` design should simplify user interaction yet remain
```

Comment:
It is a little wired to say `simplify user interaction`, maybe something like `minimize users' efforts on ...`?